### PR TITLE
Comprehensive Sub10 support

### PR DESCRIPTION
**Pull request by thomseddon**
This adds comprehensive discovery + polling support for Sub10 (recent acquired by Fastback but still distributed under the Sub10 name)

Because of the cost/market positioning (high capacity wireless backhaul) it may not be useful to the user base as a whole in which case I'm happy to just keep & maintain this branch separately :)

### CODE REVIEW

**Comment by laf** [comment]
Thanks @thomseddon. All looks good to go.

Just one thing :) We've updated the contributors agreement since you first signed it, can you take another look and submit another commit agreeing to the updated version. Just pop a space after your name to edit the file.

Thanks very much.

**Comment by laf** [comment]
@thomseddon Bump

**Comment by f0o** [comment]
@thomseddon @laf double bump :)

File: includes/polling/functions.inc.php [hunk 1]
```diff
@@ -369,8 +369,12 @@ function poll_mib_def($device, $mib_name_table, $mib_subdir, $mib_oids, $mib_gra
...
     foreach ($oidglist as $fulloid) {
         list($splitoid, $splitindex) = explode('.', $fulloid, 2);
-        if (is_numeric($snmpdata[$splitindex][$splitoid])) {
-            $fields[$oidnamelist[$oid_count]] = $snmpdata[$splitindex][$splitoid];
+        $val = $snmpdata[$splitindex][$splitoid];
+        if (is_numeric($val)) {
+            $rrdupdate .= ':'.$val;
+        }
+        elseif (preg_match("/^\"(.*)\"$/", $val, $number) && is_numeric($number[1])) {
+            $rrdupdate .= ':'.$number[1];
```
Comment (thomseddon) [inline includes/polling/functions.inc.php]: FYI - Sub10 returns some numbers wrapped like so: `"20"` so this is a workaround - this should maintain backwards compat

**Comment by thomseddon** [comment]
Added the extra line - sorry for the delay :)

**Comment by laf** [comment]
Thanks @thomseddon. All good from our side now expect the agreement, can you add the commit message as described in http://docs.librenms.org/General/Contributing/

Once done, we can merge. Sorry for being picky :/

**Comment by laf** [comment]
bump @thomseddon

**Comment by laf** [comment]
bump @thomseddon

**Comment by f0o** [comment]
bump @thomseddon

**Comment by Rosiak** [comment]
bump @thomseddon

**Comment by ekoyle** [comment]
So it looks like @thomseddon agreed to the new contributor agreement in this PR instead of opening a separate one.

**Comment by thomseddon** [comment]
Sorry for epic delay - done in #2717

**Comment by thomseddon** [comment]
Rebased against master also

**Comment by peppernicus** [comment]
This commit appears to have broken AirOS polling for wireless stats.  A debug poller sample after:

`RRD[update /opt/librenms/rrd/silonet-tom/ubnt-airmax-mib.rrd N]
ERROR: /opt/librenms/rrd/silonet-tom/ubnt-airmax-mib.rrd: expected timestamp not found in data source from N`

whereas before:

`RRD[update /opt/librenms/rrd/silonet-tom/ubnt-airmax-mib.rrd N:917:28:750:29:30:-62:34:99:-96:130000000:117000000:1:93:83]`

**Comment by laf** [comment]
@peppernicus Sorry if this has broken something for you. I actually can't see why this PR would have caused issues.

Can you provide a full output of:

./poller.php -h silonet-tom -d -r

And post the results on dpaste.de  or similar.

**Comment by peppernicus** [comment]
Not a big deal - just noticed today that overnight all of the wireless stats graphs for the handful of ubnt radios we run started reporting empty, so it was easy to find the specific commit where it began.  Thanks for being part of such a useful tool!

https://dpaste.de/FnVe

**Comment by peppernicus** [comment]
If I had to guess, the changes to includes/polling/functions.inc.php look likely to be responsible.

**Pull request merged**